pkg/operators/clustermanager/controllers/clustermanagercontroller/clustermanager_controller.go

```diff
@@ -553,6 +566,40 @@ func applyManagementResources(
 	return currentGenerations, appliedErrs, nil
 }
 
+// removeStoredVersion remove deleted api version from CRD status.StoredVersions
+func removeStoredVersion(ctx context.Context, apiExtensionClient apiextensionsclient.Interface) error {
+	for name, version := range removedCRDStoredVersions {
+		// retrieve CRD
+		crd, err := apiExtensionClient.ApiextensionsV1().CustomResourceDefinitions().Get(ctx, name, metav1.GetOptions{})
+		if errors.IsNotFound(err) {
+			continue
+		}
+		if err != nil {
+			klog.Warningf("faield to get CRD %v: %v", crd.Name, err)
+			continue
+		}
+
+		// remove v1alpha1 from its status
+		oldStoredVersions := crd.Status.StoredVersions
+		newStoredVersions := make([]string, 0, len(oldStoredVersions))
+		for _, stored := range oldStoredVersions {
+			if stored != version {
+				newStoredVersions = append(newStoredVersions, stored)
+			}
+		}
+		crd.Status.StoredVersions = newStoredVersions
+		// update the status sub-resource
+		crd, err = apiExtensionClient.ApiextensionsV1().CustomResourceDefinitions().UpdateStatus(ctx, crd, metav1.UpdateOptions{})
```

Comment:
No need to update the status of the CRD if there is no change at all.

Reply:
This function only update CRDs defined in removedCRDStoredVersions

Reply:
Oh, I see, do you mean when running sync(), no need to update the status every time?

Reply:
Fixed